packages/cards-editor-middleware/src/index.ts

```diff
@@ -0,0 +1,123 @@
+import type { RequestHandler, Request, Response } from 'express';
+import type { MiddlewareParameters } from '@ui5/server';
+import { json, Router as createRouter } from 'express';
+import path, { join } from 'path';
+import { existsSync, writeFileSync, mkdirSync, readFileSync } from 'fs';
+import { render } from 'ejs';
+import * as utils from './utilities';
+import os from 'os';
+
+export const enum ApiRoutes {
+    previewGeneratorSandbox = '/test/flpGeneratorSandbox.html',
+    cardsStore = '/cards/store',
+    i18nStore = '/editor/i18n'
+}
+
+module.exports = async ({ resources }: MiddlewareParameters<any>): Promise<RequestHandler> => {
+    const router = createRouter();
+    router.use(json());
+
+    const manifest = await resources.rootProject.byPath('/manifest.json');
+    if (!manifest) {
+        throw new Error('No manifest.json found.');
+    }
+
+    router.get(ApiRoutes.previewGeneratorSandbox, async (_req, res: Response) => {
+        const app = JSON.parse(await manifest.getString())['sap.app'];
+        res.status(200).send(
+            render(readFileSync(join(__dirname, '../templates/editor.html'), 'utf-8'), {
+                templateModel: {
+                    appTitle: app.title || 'Card Editor Preview',
+                    component: app.id
+                }
+            })
+        );
+    });
+
+    router.post(ApiRoutes.cardsStore, async (req: Request, res: Response) => {
+        try {
+            const floorplan = req.body.floorplan;
+            const localPath = req.body.localPath;
+            const fileName = req.body.fileName || 'manifest.json';
+            const multipleCards = utils.prepareCardTypesForSaving(req.body.manifests);
+            const BASE_PATH = '/webapp';
```

Comment:
Please use the `@sap-ux/project-access` module to get the `manifest.json` instead of hardcoding the `webapp` path